crates/net/network/src/manager.rs

```diff
@@ -942,28 +924,49 @@ where
         // iterations in << 100µs in most cases. On average it requires ~50µs, which is inside the
         // range of what's recommended as rule of thumb.
         // <https://ryhl.io/blog/async-what-is-blocking/>
-        let mut budget = 10;
 
-        loop {
-            // advance the swarm
-            match this.swarm.poll_next_unpin(cx) {
-                Poll::Pending | Poll::Ready(None) => break,
-                Poll::Ready(Some(event)) => this.on_swarm_event(event),
-            }
+        // poll new block imports (expected to be a noop for POS)
+        let maybe_more_block_imports = poll_nested_stream_with_yield_points!(
+            "net",
+            "Block imports stream",
+            BUDGET_ONCE,
+            this.block_import.poll_next_unpin(cx),
+            |outcome| this.on_block_import_result(outcome),
+        );
```

Comment:
this we don't need, because
this is a noop in POS and even if this is manually set we only expect a block every few secs, depending on the block time

so I'd like to keep using a simple while loop